### PR TITLE
kms: remove livez check

### LINKED ISSUE #93968: kmsv2: consider removing KMS from /livez checks
**What happened**:
kube-apiserver goes unhealthy and restarts when a kms provider returns PermissionDenied on Encrypt/Decrypt requests. This may happen when the permission to encrypt and decrypt gets revoked.

**What you expected to happen**:
I expected kube-apiserver to stay healthy and return errors on types covered by the kms provider.

**How to reproduce it (as minimally and precisely as possible)**:
Revoke permission to encrypt/decrypt data by a KMS key.

**Anything else we need to know?**:
[Code](https://github.com/kubernetes/kubernetes/blob/ed4c2dbf9211bec3e080488b0fd67a9725759a45/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L139) just does classic `err != nil` so I think that checking error code here will resolve the issue.

### PR DESCRIPTION
**Pull request by ritazh**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Remove /livez livezchecks for KMS v1 and v2 to ensure KMS health does not cause kube-apiserver restart. KMS health checks are still in place as a healthz and readiness checks.

#### Which issue(s) this PR fixes:

Fixes #93968

#### Does this PR introduce a user-facing change?

```release-note
Remove /livez livezchecks for KMS v1 and v2 to ensure KMS health does not cause kube-apiserver restart. KMS health checks are still in place as a healthz and readiness checks.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/uncc

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 1]
```diff
@@ -332,18 +332,23 @@ func (s *EtcdOptions) maybeApplyResourceTransformers(c *server.Config) (err erro
...
 		if !s.SkipHealthEndpoints {
-			c.AddHealthChecks(encryptionConfiguration.HealthChecks...)
+			s.addHealthChecks(c, encryptionConfiguration.HealthChecks...)
 		}
 	}
 
 	return nil
 }
 
+func (s *EtcdOptions) addHealthChecks(c *server.Config, healthChecks ...healthz.HealthChecker) {
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go:347]: `s` is not used and this need to indicate that `livez` is not wired.

```suggestion
func addHealthChecksWithoutLivez(c *server.Config, healthChecks ...healthz.HealthChecker) {
```

File: test/integration/controlplane/transformation/transformation_test.go [hunk 2]
```diff
@@ -550,7 +551,7 @@ func (e *transformTest) printMetrics() error {
 func mustBeHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains string, clientConfig *rest.Config, excludes ...string) {
 	t.Helper()
 	var restErr error
-	pollErr := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+	pollErr := wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) {
```
Comment (enj) [inline test/integration/controlplane/transformation/transformation_test.go:554]: What is the purpose of this change?

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go [hunk 3]
```diff
@@ -346,13 +370,13 @@ func TestReadinessCheck(t *testing.T) {
 				t.Fatalf("Failed to add healthz error: %v", err)
 			}
 
-			healthChecksAreEqual(t, tc.wantReadyzChecks, serverConfig.ReadyzChecks)
-			healthChecksAreEqual(t, tc.wantHealthzChecks, serverConfig.HealthzChecks)
+			healthChecksAreEqual(t, tc.wantReadyzChecks, serverConfig.ReadyzChecks, "readyz")
+			healthChecksAreEqual(t, tc.wantHealthzChecks, serverConfig.HealthzChecks, "healthz")
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd_test.go:374]: I would check livez here too for consistency.

**Review by enj — CHANGES_REQUESTED** [review]

**Comment by enj** [comment]
@ritazh release note should indicate that this impacts both kms v1 and kms v2 since this is a shared code path.

File: test/integration/controlplane/transformation/transformation_test.go (hunk 2 above)
Comment (ritazh) [reply test/integration/controlplane/transformation/transformation_test.go:554]: pull-kubernetes-linter-hints failed due to the deprecation of PollImmediate and suggested to use PollUntilContextTimeout

**Comment by ritazh** [comment]
> @ritazh release note should indicate that this impacts both kms v1 and kms v2 since this is a shared code path.

updated.
